xdsl/dialects/builtin.py

```diff
@@ -715,7 +716,48 @@ def print_builtin(self, printer: Printer) -> None:
         printer.print_string("loc(unknown)")
 
 
-LocationAttr = UnknownLoc
+@irdl_attr_definition
+class FileLineColLoc(ParametrizedAttribute, BuiltinAttribute):
+    """
+    Syntax:
+
+    ```
+    filelinecol-location ::= string-literal `:` integer-literal `:`
+                             integer-literal
+    ```
+
+    An instance of this location represents a tuple of file, line number, and
+    column number. This is similar to the type of location that you get from
+    most source languages.
+
+    Example:
+
+    ```mlir
+    loc("mysource.cc":10:8)
+    ```
+    """
+
+    name = "file_line_loc"
+
+    filename: StringAttr = param_def()
+    line: IntAttr = param_def()
+    column: IntAttr = param_def()
+
+    @staticmethod
+    def get(filename: str, line: int, column: int):
+        return FileLineColLoc(StringAttr(filename), IntAttr(line), IntAttr(column))
```

Comment:
why `get` instead of a custom init? Maybe I missed part of the discussion on constructors for parametrized attributes?

Reply:
I feel if we can get converters sorted then we would use an `init` right? If so then it makes sense to me to use an `init` instead of `get`. Maybe I'm missing an advantage of `get` though?

Reply:
No you're right, I'll remove the get for now and wait for converters in the future.